### ISSUE TITLE
Unable to add new values to Enum column if it belongs to primary key

### ISSUE THREAD

**Issue by valyala** [yandex/ClickHouse#364]
# The issue

We had a table with an Enum column in primary key:

```sql
CREATE TABLE x (
   EventDate Date,
   a Enum8('foo'=1)
) Engine = MergeTree(EventDate, (EventDate, a), 8192);
```

Then we tried adding new Enum value to column `a`:

```sql
ALTER TABLE x MODIFY COLUMN a Enum8('foo'=1, 'bar'=2);
```

Unfortunately, `clickhouse` returned the error instead of altering the table:

```
DB::Exception: trying to ALTER key column a
```

# Expected behavior

`clickhouse` must note that the new Enum for column `a` contains all the old values plus new one, so it must instantly add new value to `a` Enum without column scanning/rebuilding.

**Comment by ludv1x** [comment]
Ok, LGTM :+1:

**Comment by DenisCheremisov** [comment]
Happens to me as well. As far as I remember Alexey mentioned this issue is planned to be fixed.

**Comment by ztlpn** [comment]
Available in v1.1.54165
